torch/nn/utils/rnn.py

```diff
@@ -234,7 +235,10 @@ def pad_sequence(sequences, batch_first=False, padding_value=0):
     # in sequences are same and fetching those from sequences[0]
     max_size = sequences[0].size()
     trailing_dims = max_size[1:]
-    max_len = max([s.size(0) for s in sequences])
+    max_len = max((s.size(0) for s in sequences))
```

Comment:
I'm pretty sure this doesn't handle the `batch_first=True` case correctly, although it looks like our current code is broken too? Can you please fix that?